parsec/backend/invite.py

```diff
@@ -434,33 +403,67 @@ def _to_http_redirection_url(
+            # However, we'd rather limit the use of missing fields to compatibility use cases (e.g when a
+            # field has been added in a new version but does not exist in older versions). In this case, we
+            # can replace the missing field with `SUCCESS` without breaking compatibility with older clients
+            # since they also choose `SUCCESS` as value when getting an `AttributeError` on the reply.
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.SUCCESS)
+
+        # Backend address not configured, we won't be able to send the email
+        if not self._config.backend_addr:
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.NOT_AVAILABLE)
+
+        # Generate email message
+        if req.type == InvitationType.USER:
+            to_addr = invitation.claimer_email
+            if client_ctx.human_handle:
+                greeter_name = client_ctx.human_handle.label
+                reply_to = f"{client_ctx.human_handle.label} <{client_ctx.human_handle.email}>"
+            else:
+                greeter_name = client_ctx.user_id.str
+                reply_to = None
+            message = generate_invite_email(
+                from_addr=self._config.email_config.sender,
+                to_addr=invitation.claimer_email,
+                greeter_name=greeter_name,
+                reply_to=reply_to,
+                organization_id=client_ctx.organization_id,
+                invitation_url=_to_http_redirection_url(client_ctx, invitation),
+                backend_url=self._config.backend_addr.to_http_domain_url(),
+            )
+        else:  # Device
+            to_addr = client_ctx.human_handle.email
+            message = generate_invite_email(
+                from_addr=self._config.email_config.sender,
+                to_addr=client_ctx.human_handle.email,
+                greeter_name=None,
+                reply_to=None,
+                organization_id=client_ctx.organization_id,
+                invitation_url=_to_http_redirection_url(client_ctx, invitation),
+                backend_url=self._config.backend_addr.to_http_domain_url(),
+            )
+
+        # Send the email
+        try:
+            await send_email(
+                email_config=self._config.email_config,
+                to_addr=to_addr,
+                message=message,
+            )
+        except InvitationEmailRecipientError:
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.BAD_RECIPIENT)
+        except InvitationEmailConfigError:
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.NOT_AVAILABLE)
+        except Exception:
+            # Fail-safe: since the device/user has been created, we don't want to fail too hard
+            logger.exception("Unexpected exception while sending an email")
+            return InviteNewRepOk(invitation.token, InvitationEmailSentStatus.NOT_AVAILABLE)
```

Comment:
It's nice to prevent hard crash like that, but we don't have any log on the exception that may be raised here.

Reply:
`logger.exception` logs the exception in the current context

Reply:
`logger.exception` takes last exception automatically, so it's equivalent to:
```python
except Exception as exc:
    logger.error("foo", exc_info=exc)
```